File: parking_system_java/Floor.java
package parking_system;

import java.util.List;
import java.util.ArrayList;

class Floor {
    private String floorId;
    private List<Spot> spots;

    public Floor(String id, int size) {
        this.floorId = id;
        this.spots = new ArrayList<>();
        for(int i = 0; i < size; i++) {
            String key = floorId + i;
            Spot spot = new Spot(key);
            spots.add(spot);
        }
    }

    public List<Spot> getSpots() {
        return spots;
    }

    @Override
    public String toString() {
        return "Floor{" + floorId + ": " + "spots=" + spots + '}';
    }
}

File: parking_system_cpp/floor.cpp
#include "floor.h"
#include <string>
#include <vector>

Floor::Floor(const std::string& floorId, int size) : floorId(floorId) {
    for (int i = 0; i < size; i++) {
        std::string str = floorId + static_cast<char>('0' + i);
        spots.emplace_back(str);
    }
}

const std::vector<Spot>& Floor::getSpots() const {
    return spots;
}

Spot* Floor::getFirstVacantSpot() {
    for (Spot& spot : spots) {
        if (spot.isVacant()) return &spot;
    }
    return nullptr;
}

std::string Floor::to_string() const {
    std::string res = "Floor ID: " + floorId + "\nSpots:\n";
    for (const auto& spot : spots) {
        res += "  - " + spot.to_string() + "\n";
    }
    return res;
}
